langwatch_nlp/langwatch_nlp/logger.py

```
@@ -1,20 +1,163 @@
+
 import logging
+import os
+import sys
+from contextvars import ContextVar
+from typing import Any, Optional
+
+import structlog
+from structlog.typing import EventDict, WrappedLogger
+
+# Context variable for request-scoped data (like project_id)
+_log_context: ContextVar[dict[str, Any]] = ContextVar("log_context", default={})
```

Ruff (0.14.14)

[warning] Do not use mutable data structures for `ContextVar` defaults

(B039)